metrics/xrootd_metrics.go

```diff
@@ -400,23 +400,27 @@ func ConfigureMonitoring(ctx context.Context, egrp *errgroup.Group) (int, error)
 		return nil
 	})
 
-	go func() {
-		var buf [65536]byte
-		for {
-			// TODO: actually parse the UDP packets
-			plen, _, err := conn.ReadFromUDP(buf[:])
-			if errors.Is(err, net.ErrClosed) {
-				return
-			} else if err != nil {
-				log.Errorln("Failed to read from UDP connection", err)
-				continue
-			}
-			PacketsReceived.Inc()
-			if err = HandlePacket(buf[:plen]); err != nil {
-				log.Errorln("Failed to handle packet:", err)
+	// if the shoveler is not enabled, then we need to listen to the UDP packets coming from XRootD
+	// if the shoveler is enabled, then the shoveler will listen to XRootD and update the metrics
+	if !param.Shoveler_Enable.GetBool() {
+		go func() {
+			var buf [65536]byte
+			for {
+				// TODO: actually parse the UDP packets
+				plen, _, err := conn.ReadFromUDP(buf[:])
+				if errors.Is(err, net.ErrClosed) {
+					return
+				} else if err != nil {
+					log.Errorln("Failed to read from UDP connection", err)
```

Comment:
Same gist here -- give this a bit more context, e.g. "Failed to read from UDP connection while aggregating xrootd metrics."